universal-application-tool-0.0.1/app/services/program/ProgramDefinition.java

```diff
@@ -186,6 +186,13 @@ public boolean hasQuestion(long questionId) {
     return questionIds.get().contains(questionId);
   }
 
+  /** Get the block definitions associated with the repeater id. */
+  public ImmutableList<BlockDefinition> getBlockDefinitions(Optional<Long> repeaterId) {
```

Comment:
Nit (blocking): the function name is a little bit confusing if without the context of the parameters. Would you call this `getRepeaterBlockDefinitions`?

Reply:
I totally agree. I want this method to also work with `Optional.empty` to get all non-repeated block definitions. 

Any ideas on naming? I am SO bad at names.

Reply:
I am also very bad at naming. Does `getBlockDefinitionsForRepeater` sound better? And in the docstring says `Optional#empty` means `does not repeat`.

Reply:
>  I want this method to also work with Optional.empty to get all non-repeated block definitions.

There's already `blockDefinitions()` for that. Recommend changing to `getBlockDefinitionsForRepeater(Long repeaterId)`.

Reply:
Done! I added a `getNonRepeatedBlockDefinitions` method too.